packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -735,18 +749,19 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
             /* @conditional-compile-remove(mention) */
             setInputTextValue(newValue ?? '');
             /* @conditional-compile-remove(mention) */
-            handleOnChange(
-              e,
+            handleOnChange({
+              event: e,
               tagsValue,
-              textValue,
+              htmlTextValue: textValue,
               inputTextValue,
               currentTriggerStartIndex,
-              selectionStartValue === null ? undefined : selectionStartValue,
-              selectionEndValue === null ? undefined : selectionEndValue,
-              e.currentTarget.selectionStart === null ? undefined : e.currentTarget.selectionStart,
-              e.currentTarget.selectionEnd === null ? undefined : e.currentTarget.selectionEnd,
-              newValue
-            );
+              previousSelectionStart: selectionStartValue === null ? undefined : selectionStartValue,
+              previousSelectionEnd: selectionEndValue === null ? undefined : selectionEndValue,
+              currentSelectionStart:
+                e.currentTarget.selectionStart === null ? undefined : e.currentTarget.selectionStart,
```

Comment:
It may be a good idea to add a func to help with this check (if `null` -> use `undefined`) to have it in 1 place only